lib/usdUfe/ufe/UsdClipboardCommands.cpp

```diff
@@ -0,0 +1,496 @@
+    PXR_NS::SdfLayerRefPtr layer = PXR_NS::SdfLayer::CreateAnonymous();
+    PXR_NS::UsdStageRefPtr clipboardStage
+        = PXR_NS::UsdStage::Open(layer->GetIdentifier(), PXR_NS::UsdStage::LoadNone);
+
+    if (!clipboardStage) {
+        // It shouldn't be possible to obtain an invalid stage since we are creating it from an
+        // anonymous layer, however as a precaution we leave this check.
+        const std::string error = "Failed to create Clipboard stage.";
+        throw std::runtime_error(error);
+    }
+
+    // Duplicate the selected items to the Clipboard stage using its pseudo-root as parent
+    // item destination.
+    auto usdParentItem = UsdSceneItem::create(Ufe::Path(), clipboardStage->GetPseudoRoot());
+    auto duplicateSelectionUndoableCmd
+        = UsdUndoDuplicateSelectionCommand::create(_selection, usdParentItem);
+    duplicateSelectionUndoableCmd->execute();
+
+    // Set Clipboard metadata.
+    setClipboardMetadata(duplicateSelectionUndoableCmd);
+
+    // Set the clipboard data.
+    _clipboard->setClipboardData(clipboardStage);
+}
+
+void UsdCopyClipboardCommand::undo()
+{
+    // Do nothing.
+}
+
+void UsdCopyClipboardCommand::redo()
+{
+    // Do nothing.
+}
+
+// Ensure that UsdCutClipboardCommand is properly setup.
+static_assert(std::is_base_of<Ufe::UndoableCommand, UsdCutClipboardCommand>::value);
+static_assert(std::has_virtual_destructor<UsdCutClipboardCommand>::value);
+static_assert(!std::is_copy_constructible<UsdCutClipboardCommand>::value);
+static_assert(!std::is_copy_assignable<UsdCutClipboardCommand>::value);
+static_assert(!std::is_move_constructible<UsdCutClipboardCommand>::value);
+static_assert(!std::is_move_assignable<UsdCutClipboardCommand>::value);
+
+UsdCutClipboardCommand::UsdCutClipboardCommand(
+    const Ufe::Selection&    selection,
+    const UsdClipboard::Ptr& clipboard)
+    : _selection(selection)
+    , _clipboard(clipboard)
+{
+}
+
+UsdCutClipboardCommand::Ptr
+UsdCutClipboardCommand::create(const Ufe::Selection& selection, const UsdClipboard::Ptr& clipboard)
+{
+    if (selection.empty()) {
+        return {};
+    }
+
+    return std::make_shared<UsdCutClipboardCommand>(selection, clipboard);
+}
+
+void UsdCutClipboardCommand::execute()
+{
+    UsdUndoBlock undoBlock(&_undoableItem);
+
+    // Step 1. Copy the selected items to the Clipboard.
+    auto copyClipboardCommand = UsdCopyClipboardCommand::create(_selection, _clipboard);
+    copyClipboardCommand->execute();
+
+    const auto& sceneItemOpsHandler
+        = Ufe::RunTimeMgr::instance().sceneItemOpsHandler(UsdUfe::getUsdRunTimeId());
+
+    // Step 2. Delete the selected items.
+    for (auto&& item : _selection) {
+        if (UsdSceneItem::Ptr usdItem = std::dynamic_pointer_cast<UsdSceneItem>(item)) {
+            sceneItemOpsHandler->sceneItemOps(usdItem)->deleteItem();
+        }
+    }
+}
+
+void UsdCutClipboardCommand::undo() { _undoableItem.undo(); }
```

Comment:
This assumes all items in the selection were USD items, what happens if there were Maya items too? As I understand it, sceneItemOpsHandler->sceneItemOps(usdItem)->deleteItem(); above will have deleted the Maya items?

Reply:
In a mixed Ufe selection, Maya will only cut (copy then delete) the Maya nodes and Usd will only cut the prims. You'll see that the command is given a Ufe::Selection - this contains only usd items.